### PR TITLE
fix: [PIE-2792]: azure connector setup

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:

Azure Connector setup 

##### Jira Links:

https://harness.atlassian.net/browse/PIE-2792 credential flow
https://harness.atlassian.net/browse/PIE-3214 managed identity flow
cd-ng azure service setup 

##### Screenshots:
https://user-images.githubusercontent.com/96610348/161084489-316b6338-2311-422b-9df0-23c4a3a78cde.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by satyam-harness — DISMISSED** [review]

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+          setLoadingConnectorSecrets(false)
+        })
+      } else {
+        setLoadingConnectorSecrets(false)
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingConnectorSecrets])
+
+  // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:133]: Let's avoid these comments as much as possible.

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+  const handleSubmit = (formData: ConnectorConfigDTO) => {
+    nextStep?.({ ...props.connectorInfo, ...prevStepData, ...formData } as StepConfigureProps)
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:138]: Please give type to formik param.

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+      case AzureSecretKeyType.SECRET:
+        formik.setFieldValue('secretText', undefined)
+        return
+      case AzureSecretKeyType.CERT:
+        formik.setFieldValue('secretFile', undefined)
+        return
+    }
+  }
+
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:150]: Same as above comment.

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+    await act(async () => {
+      fireEvent.change(container.querySelector('input[name="name"]')!, {
+        target: { value: 'dummy name' }
+      })
+    })
+    expect(container).toMatchSnapshot()
+
+    await act(async () => {
+      fireEvent.click(container.querySelector('button[type="submit"]')!)
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:198]: Find the button by using **getByText** or **findByText**

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx (hunk 1 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:133]: will try to remove this eslint disablers, thanks :)

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx (hunk 2 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:138]: will provide the type, thanks :)

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx (hunk 3 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/StepAuth/AzureAuthentication.tsx:150]: will provide the type, thanks :)

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx (hunk 4 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:198]: will replace it, thanks :)

**Comment by Tamaraius** [comment]
Closed due to change in develop, creating a new PR

**Pull request closed without merging**